### PR TITLE
wordpress integration guide

### PR DESCRIPTION
**Pull request by richardesigns**
Hi Peter, hope you like this.  It's quite screenshotty but I'm going with the idea that the average Wordpress developer has fewer programming skills so I thought we should be as helpful as possible.

I can't seem to work out how to get the guide to list but the page works if you visit the link:

localhost:4000/guides/wordpress-openid-connect-client-plugin-integration/

### CODE REVIEW

**Comment by CrowdHailer** [comment]
> I can't seem to work out how to get the guide to list but the page works if you visit the link:

Just search for the word phoenix and add it next to the existing links. e.g in `docs/index.md` and `_layout/doc.html`

docs

**Comment by CrowdHailer** [comment]
None of these a blocker but I would suggest trying to keep even closer to the format of the other.

- Use the same help layout as other articles, i.e. bottom only. I didn't offer help at the top of the article but at the bottom, readers probably want the specifics asap, and if they are looking for help at the beginning of the article have probably looked for a contact us page or something else
- Use the same type of headers as on the other articles, i.e. they are higher numbered headers
- Use the same format for headers as the other articles i.e. don't write "step 1"
- use a similar requirements section as other articles, most importantly make sure there is a link to a wordpress home page or getting started with wordpress page
- I tried to put the bit about setting up an account with DID last so the article was easier to repost in otherplaces under the guise of just explaining about OpenID Connect
- This is an unhelpful line "Leave the ‘Test mode’ box unchecked." what if they are building a test site?
- To my eyes I would just use the same "Setup the App on DID" section from other articles, but with the correct screenshot. I think writing down click the button marked create app is just bloat. that page should be as explanatory as possible because some users get to that page without having a guide open next to them.


Consistency is important. I don't know why you have chosen to do them differently but if you think it is worth adopting a new style then we should change the other articles.
I would suggest using the same section on creating an app on DID to save time, so now you have written a different one that time can't be saved but potentially using exactly the same one might save time in the future because we can then use find and replace when need to make changes to all the articles

File: guides/wordpress-openid-connect-client-plugin-integration/index.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+Select ‘Custom OpenID Connect App’.  You will now see this screen:
+
+![screenshot of miniOrange OpenId connect plugin config dashboard](/assets/images/wordpress-guide/plugin-config.jpg)
+
+Complete the fields using the following values:
+
+```
+App Name: DID
+Client ID: *copy and paste this from your DID app settings page.*
```
Comment (CrowdHailer) [inline guides/wordpress-openid-connect-client-plugin-integration/index.md]: It's a shame the styling doesn't look quite right here.
Could it just show these values written into the correct boxes in the screenshot?

File: guides/wordpress-openid-connect-client-plugin-integration/index.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+![screenshot of adding login buttons to Wordpress template content blocks](/assets/images/wordpress-guide/widget1.PNG)
+
+## Optional: Login Button Styling
+
+The ‘Login with DID’ button can be styled with some simple CSS rules.  The following CSS can be used to style the button with DID’s colour scheme:
+
+```
+.mo_oauth_login_button {
+display: block;
```
Comment (CrowdHailer) [inline guides/wordpress-openid-connect-client-plugin-integration/index.md]: Can you fix the indentation on this code block.

File: guides/wordpress-openid-connect-client-plugin-integration/index.md (hunk 1 above)
Comment (CrowdHailer) [reply guides/wordpress-openid-connect-client-plugin-integration/index.md]: the other option is to have it as a list
```
- **App Name:** DID
```

**Comment by CrowdHailer** [comment]
@richardesigns I think it looks good, good to see all the steps written down and that there really isn't all that many.

Sorry if my first comment was a bit moany about doing it my way. I think consistency is important and doing things like copying the sections that are the same means we can basically skip over, checking them.

**Comment by richardesigns** [comment]
All fine, hadn't spotted the h2 / h3.  Fixed that.

Agree with your points.

I've been slightly more descriptive with the 'create an app' section but have thinned it too.  I think it does need to be in the flow where i have it rather than at the end because this is more of a 'step by step' hand holding guide.  I suspect there is room for causing confusion if we say 'copy your client ID' in here and the user is like 'what client id'?  

I do feel it needs to be in the right order rather than at the end but I can also see why it would go at the end for an 'advanced' guide like the Nodejs one.  

Slightly different audience / developer type.

File: guides/wordpress-openid-connect-client-plugin-integration/index.md (hunk 2 above)
Comment (richardesigns) [reply guides/wordpress-openid-connect-client-plugin-integration/index.md]: yes absolutely

File: guides/wordpress-openid-connect-client-plugin-integration/index.md (hunk 1 above)
Comment (richardesigns) [reply guides/wordpress-openid-connect-client-plugin-integration/index.md]: list

File: guides/wordpress-openid-connect-client-plugin-integration/index.md [hunk 3]
```diff
@@ -34,7 +26,21 @@ You can also visit the plugin’s page to download, install and activate manuall
...
-## Step 3. Configure the plugin
+### Setup the App on did
+
+You will need a DID account. [Sign up]({{ site.auth_origin }}) to create one now.
+
+After signing up, you will be directed to set up your first app.
+
+Give your new app a descriptive name e.g. ‘Wordpress Demo’ and use the full URL of the website in the ‘host’ box.
+
+Leave the ‘Test mode’ box unchecked to use this app on a live Wordpress website. Test mode can be used for localhost installations.
```
Comment (CrowdHailer) [inline guides/wordpress-openid-connect-client-plugin-integration/index.md]: Test mode isn't just for localhost. If you have a staging or example site on the internet you can use Test their too

**Comment by CrowdHailer** [comment]
>  I suspect there is room for causing confusion if we say 'copy your client ID' in here and the user is like 'what client id'?
>
> I do feel it needs to be in the right order rather than at the end but I can also see why it would go at the end for an 'advanced' guide like the Nodejs one.

If this is true that this is causing confusion in the node article then we need to fix it. Node.js are not the advanced community. In fact i suspect the average node developer uses node because they did front end js and have just decided to just try and make that knowledge stretch a bit further for them.


Anyways probably not worth doing now. 
Merge away and crost post and all that jazz. I wonder if we can get any promotion from mini orange.
Actually once we merge we could ask them to look at the article

File: guides/wordpress-openid-connect-client-plugin-integration/index.md (hunk 3 above)
Comment (richardesigns) [reply guides/wordpress-openid-connect-client-plugin-integration/index.md]: ah coool ok, I'll leave this line out.  Perhaps somewhere else we can discuss what the test mode is for and how it works.

**Pull request merged**